File: AIT SOLUTION/backend/app/schemas/airplane.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class AirplaneBase(BaseModel):
    model: str
    registration: str
    manufacturer: str
    total_seats: int
    economy_seats: int
    premium_economy_seats: int = 0
    business_seats: int = 0
    first_class_seats: int = 0
    rows_count: int
    seats_per_row: int

class AirplaneCreate(AirplaneBase):
    pass

class AirplaneRead(AirplaneBase):
    id: int
    owner_id: Optional[int] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

File: AIT SOLUTION/mobile_app/lib/features/airplane/domain/entities/airplane.dart
import 'package:equatable/equatable.dart';

class Airplane extends Equatable {
  final int id;
  final String model;
  final String registration;
  final String manufacturer;
  final int totalSeats;
  final int economySeats;
  final int businessSeats;

  const Airplane({
    required this.id,
    required this.model,
    required this.registration,
    required this.manufacturer,
    required this.totalSeats,
    required this.economySeats,
    this.businessSeats = 0,
  });

  @override
  List<Object?> get props => [
    id, model, registration, manufacturer, 
    totalSeats, economySeats, businessSeats
  ];
}
